system/Image/Image.php

```diff
@@ -0,0 +1,163 @@
+{
+    protected $driver;
+    protected $resource;
+    protected $imageFullPath;
+    protected $imageFolder;
+    protected $imageFilename;
+    protected $imageExtension;  
+    protected $mime;
+    
+    protected $imageTypes = [
+            'jpg',
+            'jpeg',
+            'png',
+            'gif',
+            'tiff'
+    ];
+    
+    // ------------------------------------------------------------------
+    
+    /**
+     * @param string $driver
+     * @return resource
+     */
+    public function load(string $driver): resource
+    {
+        $classname = "Codeigniter\\Image\\$driver";
+        return new $classname;
+    }
+    
+    // ------------------------------------------------------------------
+    
+    /**
+     * @param string $folder
+     * @return resource
+     */
+    public function setFolder(string $folder): resource
+    {
+        $this->imageFolder = $folder;
+        return $this;
+    }
+    
+    // ------------------------------------------------------------------
+    
+    /**
+     * @param string $filename
+     * @return resource
+     */
+    public function setFilename(string $filename): resource
+    {
+        $this->imageFilename = $filename;
+        return $this;
+    }
+    
+    // ------------------------------------------------------------------
+    
+    /**
+     * @param string $extension
+     * @throws \InvalidArgumentException
+     * @return resource
+     */
+    public function setExtension(string $extension): resource
+    {
+        if (in_array($extension, $this->imageTypes))
+        {
+            $this->imageExtension = strtolower($extension);
+            return $this;
+        }
+        else
+        {
+            throw new \InvalidArgumentException('Image extension is not allowed');
+        }
+    }
+    
+    // ------------------------------------------------------------------
+    
+    /**
+     * @param string $full_image_path
+     * @throws \InvalidArgumentException
+     * @return resource
+     */
+    public function image(string $full_image_path = false, $driver): resource
+    {
+        if ($full_image_path)
+        {
+            $route = pathinfo($full_image_path);
+            $this->imageFilename = $route['filename'];
+            $this->imageFolder = $route['dirname'];
+            $extension = $rute['extension'];
+            
+            if (in_array($extension, $this->imageTypes))
+            {
+                $this->imageExtension = strtolower($extension);
+            }
+            else
+            {
+                throw new \InvalidArgumentException('Image extension is not allowed');
+            }
+        }
+        
+        $this->imageFullPath = BASEPATH.$this->imageFolder.'/'.$this->imageFilename.'.'.
+                $this->imageExtension;
+        
+        $this->mime = "image/$this->imageExtension";
+        
+        if(file_exists($this->imageFullPath))
+        {
+            header('content_type :'.$this->mime);
```

Comment:
Why are we setting a header here? That only matters if we're sending the file to the browser. If we're only modifying a local file, this is pointless and could lead to inaccurate  headers being sent. Additionally, it has issues with content/header order errors, and, if it was desired, should be using the Response object for consistency and safety.